controllers/ProfileController.js

```diff
@@ -13,69 +13,97 @@ export default class ProfileController {
-      bio,
-    });
-    const { firstName, lastName } = req.user;
-    const resData = {
-      message: 'Your profile has been created successfully',
-      profile: {
-        name: `${firstName} ${lastName}`,
-        bio,
+  static async createProfile(req, res, next) {
+    try {
+      const { id } = req.user;
+      const { dataValues } = await models.User.findOne({ where: { id } });
+      if (!dataValues.isVerified) {
+        return errorResponse(res, 401, 'You need to verify your account first');
+      }
+      const { avatar, bio } = req.body;
+      const profileExist = await models.Profile.findOne({ where: { userId: id } });
+      if (profileExist) {
+        return errorResponse(res, 409, 'Profile already exists');
+      }
+      await models.Profile.create({
+        userId: id,
         avatar,
-      },
-    };
-    return successResponse(res, 201, resData);
+        bio,
+      });
+      const { firstName, lastName } = req.user;
+      const resData = {
+        message: 'Your profile has been created successfully',
+        profile: {
+          name: `${firstName} ${lastName}`,
+          bio,
+          avatar,
+        },
+      };
+      return successResponse(res, 201, resData);
+    } catch (error) {
+      next(error);
+    }
   }
 
   /**
    *
    * @method editProfile
    * @param {object} req express request object
    * @param {object} res express response object
+   * @param {function} next
    * @returns {object} returns profile info
    */
-  static async editProfile(req, res) {
-    const {
-      avatar, bio, firstName, lastName,
-    } = req.body;
-    let { id } = req.params;
-    id = parseInt(id, 10);
-    if (id !== req.user.id) {
-      return errorResponse(res, 403, 'You are not allowed to edit this profile');
-    }
-    await models.Profile.update(
-      { bio, avatar },
-      { where: { userId: id } },
-    );
+  static async editProfile(req, res, next) {
+    try {
+      let { id } = req.params;
+      const hasProfile = await models.Profile.findOne({ where: { userId: id } });
+
+      id = parseInt(id, 10);
+      if (id !== req.user.id) {
+        return errorResponse(res, 403, 'You are not allowed to edit this profile');
+      }
 
-    await models.User.update(
-      { firstName, lastName },
-      { where: { id } },
-    );
-    const { dataValues } = await models.User.findOne({ where: { id } });
-    const resData = {
-      message: 'Your profile has been updated successfully',
-      profile: {
+      if (!hasProfile) {
+        return errorResponse(res, 404, 'You don\'t have a profile, please create a profile');
+      }
+
+      const {
         avatar,
         bio,
-        firstName: dataValues.firstName,
-        lastName: dataValues.lastName,
-      },
-    };
-    return successResponse(res, 200, resData);
+        firstName,
+        lastName,
+      } = req.body;
+
+      await models.Profile.update(
+        { bio, avatar },
+        { where: { userId: id } },
```

Comment:
1. Must a user provide a new bio and avatar before this update can take place?
2. What if I want to update my name?

Reply:
No you don't need to provide your full update. If you dont provide any data, it would work fine and retain your old data